### ISSUE TITLE
Overriding a field with a getter (but no setter) causes a runtime error

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/dev_compiler#380]
This is another manifestation of overriding fields (see #52).  This particular one is a common pattern in Angular.

```dart
class A {
  String x = "hello";
}

class B extends A {
  String get x => "world";
}

void main() {
  var a = new B();
  print(a.x);
}
```

At runtime, this triggers:

```
Uncaught TypeError: Cannot set property x of [object Object] which has only a getter
```

in `A`'s constructor.

**Comment by jmesserly** [comment]
Yeah, as noted in https://github.com/dart-lang/dev_compiler/issues/52#issuecomment-94858673, the only pattern we support is using a field to implement a getter/setter. You can never override a field, either with another field, or a getter/setter (those two cases are the same underlying issue.)

So, as discussed in #52, we have two options: either it's an error and we need to patch Angular, because the current code doesn't work in Strong Mode, or we take a big codegen/readability hit on (at least) all public fields & some set of private fields.

**Comment by jmesserly** [comment]
There's some more middle of the road options we could consider. I'll move discussion of that over to #52.
